include/Dialect/Poly/IR/PolyOps.td

```diff
@@ -101,17 +75,32 @@ def Poly_GetCoeffOp : Poly_Op<"get_coeff"> {
 }
 
 def Poly_ExtractSliceOp : Poly_Op<"extract_slice"> {
-  let summary = "Extracts a slice of coefficients from X^i to X^j inclusive.";
+  let summary = "Extracts a slice of coefficients from X^i to X^j.";
+
+  let description = [{
+    Extracts a slice of coefficients from X^i to X^j.
+
+    The included range includes `i`, but excludes `j`, similar to array slice
+    semantics.
 
+    The coefficients are returned in the order they appear in the polynomial,
+    from smallest degree to largest degree
+
+  }];
   let arguments = (ins
-    Poly:$x,
+    Polynomial:$x,
     Index:$i,
     Index:$j
   );
 
   let results = (outs
     TensorOf<[AnyInteger]>:$output
   );
+
+  let builders = [
+    // Builder to infer output type from the input type
+    OpBuilder<(ins "::mlir::Value":$input, "::mlir::Value":$i, "::mlir::Value":$j)>
+  ];
```

Comment:
What about using InferTypeOpInterface instead? That way, you only have to implement the actual type inference, with the builder(s) still being auto generated? See https://github.com/MarbleHE/HECO/blob/main/include/heco/IR/FHE/FHE.td#L135-L138